### PR TITLE
prov/rxm: Static connections support

### PR DESCRIPTION
**Pull request by alexander-sannikov**
Signed-off-by: Sannikov, Alexander <alexander.sannikov@intel.com>

### CODE REVIEW

**Comment by alexander-sannikov** [comment]
@dmitrygladkov Could you please take a look.

File: prov/rxm/src/rxm_init.c [hunk 1]
```diff
@@ -367,11 +368,16 @@ RXM_INI
 			"(default: 128). Setting this to 0 would get default "
 			"value defined by the MSG provider.");
 
+	fi_param_define(&rxm_prov, "static_connect", FI_PARAM_BOOL,
+			"Enables static connections on RXM level. (default: off) "
+			"May increase startup time in case of lare run.");
```
Comment (dmitrygladkov) [inline prov/rxm/src/rxm_init.c]: `"... in case of large run."`

File: prov/rxm/src/rxm_av.c [hunk 2]
```diff
@@ -89,6 +89,13 @@ rxm_av_insert_cmap(struct fid_av *av_fid, const void *addr, size_t count,
 					"Unable to update CM for OFI endpoints\n");
 				return ret;
 			}
+			if (rxm_static_connect) {
+				struct rxm_cmap_handle *handle;
+				rxm_ep->cmap->acquire(&(rxm_ep->cmap->lock));
+				handle = rxm_cmap_acquire_handle(rxm_ep->cmap, fi_addr_tmp);
+				ret = rxm_cmap_handle_connect(rxm_ep->cmap, fi_addr_tmp, handle);
```
Comment (dmitrygladkov) [inline prov/rxm/src/rxm_av.c]: `(void) rxm_cmap_handle_connect(rxm_ep->cmap, fi_addr_tmp, handle);`

**Comment by alexander-sannikov** [comment]
Closed as a subset of PR #4692

**Pull request closed without merging**